python/cudf/cudf/utils/dtypes.py

```diff
@@ -130,80 +128,27 @@ def is_datetime_dtype(obj):
     return "M8" in obj.str
 
 
-def is_categorical_dtype(obj):
-    """Infer whether a given pandas, numpy, or cuDF Column, Series, or dtype
-    is a pandas CategoricalDtype.
-    """
-    if obj is None:
-        return False
-    if isinstance(obj, cudf.CategoricalDtype):
-        return True
-    if obj is cudf.CategoricalDtype:
-        return True
-    if isinstance(obj, np.dtype):
-        return False
-    if isinstance(obj, CategoricalDtype):
-        return True
-    if obj is CategoricalDtype:
-        return True
-    if obj is CategoricalDtypeType:
-        return True
-    if isinstance(obj, str) and obj == "category":
-        return True
-    if isinstance(
-        obj,
-        (
-            CategoricalDtype,
-            cudf.core.index.CategoricalIndex,
-            cudf.core.column.CategoricalColumn,
-            pd.Categorical,
-            pd.CategoricalIndex,
-        ),
-    ):
-        return True
-    if isinstance(obj, np.ndarray):
-        return False
-    if isinstance(
-        obj,
-        (
-            cudf.Index,
-            cudf.Series,
-            cudf.core.column.ColumnBase,
-            pd.Index,
-            pd.Series,
-        ),
-    ):
-        return is_categorical_dtype(obj.dtype)
-    if hasattr(obj, "type"):
-        if obj.type is CategoricalDtypeType:
-            return True
-    return pd.api.types.is_categorical_dtype(obj)
-
-
-def is_list_dtype(obj):
-    return (
-        type(obj) is cudf.core.dtypes.ListDtype
-        or obj is cudf.core.dtypes.ListDtype
-        or type(obj) is cudf.core.column.ListColumn
-        or obj is cudf.core.column.ListColumn
-        or (isinstance(obj, str) and obj == cudf.core.dtypes.ListDtype.name)
-        or (hasattr(obj, "dtype") and is_list_dtype(obj.dtype))
-    )
-
-
 def cudf_dtype_from_pydata_dtype(dtype):
     """ Given a numpy or pandas dtype, converts it into the equivalent cuDF
         Python dtype.
     """
-
+    if isinstance(dtype, cudf.Generic):
+        return dtype.__class__
+    if inspect.isclass(dtype):
+        if issubclass(dtype, cudf.Generic):
+            return dtype
     if is_categorical_dtype(dtype):
         return cudf.core.dtypes.CategoricalDtype
     elif dtype in cudf._lib.types.np_to_cudf_types:
         return dtype.type
     elif np.issubdtype(dtype, np.datetime64):
         dtype = np.datetime64
 
-    return infer_dtype_from_object(dtype)
+    result = cudf.dtype(infer_dtype_from_object(dtype))
+    if isinstance(result, cudf.Generic):
+        return result.__class__
+    elif inspect.isclass(result):
+        return result
 
 
 def is_scalar(val):
```

Comment:
This is now a duplicate of the other `isscalar` function we have. Can we consolidate the two?